kore/src/Kore/Builtin/Int.hs

```diff
@@ -325,3 +331,31 @@ builtinFunctions =
     log2 n
         | n > 0 = Just (smallInteger (integerLog2# n))
         | otherwise = Nothing
+
+evalEq :: Builtin.Function
+evalEq resultSort arguments@[_intLeft, _intRight] =
+    concrete <|> symbolicReflexivity
+  where
+    mkCeilUnlessDefined termLike
+      | TermLike.isDefinedPattern termLike = Condition.topOf resultSort
```

Comment:
We actually need a stronger condition here: the arguments must be `functional`. If the arguments could take multiple values, then we would have to return multiple values. Consider:

```
\or(1, 2) ==Int 1 = \or(true, false)
```